### PR TITLE
XComIdleAnimationStateMachine

### PR DESCRIPTION
**Pull request by MrNiceUK**
#269

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComIdleAnimationStateMachine.uc [hunk 1]
```diff
@@ -2016,7 +2076,8 @@ begin:
 		if( bForceDesiredCover || bForceTurnTarget )
 		{
 			`log("Starting TurnTowardsPosition towards TargetLocation"@TargetLocation@" Rotator: "@Unit.Rotation@" Vector:"@vector(Unit.Rotation), `CHEATMGR.MatchesXComAnimUnitName(Unit.Name), 'XCom_Anim');
-			TurnTowardsPosition(bForceTurnTarget ? TempFaceLocation : TargetLocation);//Latent turning function on XGUnit
+			// Single line for Issue #269
+			TurnTowardsPosition(bForceTurnTarget ? TempFaceLocation : AimAtLocation);//Latent turning function on XGUnit
 			`log("Finished TurnTowardsPosition towards TargetLocation"@TargetLocation@" Rotator: "@Unit.Rotation@" Vector:"@vector(Unit.Rotation), `CHEATMGR.MatchesXComAnimUnitName(Unit.Name), 'XCom_Anim');
```
Comment (Musashi1584) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComIdleAnimationStateMachine.uc]: might want to update the log line here too with `AimAtLocation`

Comment (MrNiceUK) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XComIdleAnimationStateMachine.uc]: Fair point. There were several points in the code in log lines didn't match the actual code already though!

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComIdleAnimationStateMachine.uc [hunk 2]
```diff
@@ -781,17 +806,23 @@ event bool SetTargetUnit()
...
+				NewTargetLocation = ExitCoverAction.TargetLocation;
 			}
 		}
 		else
 		{
 			`log("          *ExitCover action has no primary target, aiming at"@ExitCoverAction.TargetLocation, `CHEATMGR.MatchesXComAnimUnitName(Unit.Name), 'XCom_Anim');
-			NewTargetLocation = ExitCoverAction.AimAtLocation;
+			// Issue #269 These two ExitCoverAction properties are the principle reason for introducing AimAtLocation
+				AimAtLocation = ExitCoverAction.AimAtLocation;
+				NewTargetLocation = ExitCoverAction.TargetLocation;
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComIdleAnimationStateMachine.uc]: Indentation looks wrong

Comment (MrNiceUK) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XComIdleAnimationStateMachine.uc]: Right, would have come from being copy/pasted from lines 815-816

**Comment by MrNiceUK** [comment]
As discussed with robo, refactored to avoid adding properties to a native class. Also, a pesky typo meant there was a £ instead of a E in a comment...

**Pull request merged**